03.bowling/bowling.rb

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require 'rubygems'
+
+obtained_score = ARGV[0]
+scores = obtained_score.split(',')
+shots = []
+
+scores.each do |score|
+  if score == 'X'
+    shots << 10
+    shots << 0
+  else
+    shots << score.to_i
+  end
+end
+
+frames = shots.each_slice(2).to_a
+
+point = 0
+frames.each_with_index do |current_frame, i|
+  next_frame = frames[i + 1]
+  after_next_frame = frames[i + 2]
+  strike = current_frame[0] == 10
+  spare = current_frame.sum == 10
+
+  point +=
+    if strike
+      if next_frame[0] == 10
+        current_frame[0] + next_frame[0] + after_next_frame[0]
+      else
+        current_frame[0] + next_frame.sum
+      end
+    elsif spare
+      current_frame.sum + next_frame[0]
```

Comment:
```suggestion
        20 + after_next_frame[0]
      else
        10 + next_frame.sum
      end
    elsif spare
      10 + next_frame[0]
```

好みの問題ではありますが、直接10や20と書いた方が「この変数の中身なんだろう？」と考えずに済むぶん、読み手に優しいかもしれません。